crates/firewheel-core/src/node.rs

```diff
@@ -124,6 +134,7 @@ pub trait AudioNode {
         &self,
         configuration: &Self::Configuration,
         stream_info: &StreamInfo,
+        custom_state: &mut Option<Box<dyn Any>>,
```

Comment:
Similarly, could we maybe provide a newtype wrapper around the option? If there are use-cases enabled by the raw underlying type, then this can be ignored, but I think it would be nice to refine the argument down to something like:

```rs
pub struct CustomState<'a>(&'a mut Option<Box<dyn Any>>);

impl CustomState<'_> {
    // By consuming the type, we enforce that the state can only
    // be inserted once, which is kinda nice.
    pub fn insert<T: 'static>(self, state: T) {
        self.0 = Some(Box::new(state));
    }
}
```

Maybe this is over-engineering, but it could help clarify what you're supposed to do with it and maybe future-proofs the API a touch.

Reply:
I've improved it with a new ConstructProcessorContext type.